specification/k3po.control/SPEC.md

```diff
@@ -0,0 +1,195 @@
+# K3po Control Protocol
+
+## Abstract
+
+This document defines the K3po control protocol used to communicate with a remote K3po server.
+
+## Requirements
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY",
+and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://tools.ietf.org/html/rfc2119).
+
+## The Protocol
+
+The protocol is made up of Commands and Events.  Commands are actions that can be triggered by an external entity.
+Events are triggered by observed actions or state changes and are sent back to the external entity.  The protocol 
+is formatted by
+
+TODO normalize
+
+```
+Command / Event Name\n
```

Comment:
For normalization add something like

```
Control Message   =   command-line
                      *( header-field LF )
                      LF
                      [ message-body ]
```

```
command-line      = command LF
command              = token
```

```
header-field = field-name ":" field-value
```

`message-body`, `field-name`, `field-value`, and `token` are defined in [HTTP RFC](https://tools.ietf.org/html/rfc7230)